e2e/specs/appointments.spec.ts

```diff
@@ -0,0 +1,118 @@
+    await page.getByLabel('Select location').selectOption('Mobile Clinic');
+  });
+
+  await test.step('And I select “Outpatient Department” service', async () => {
+    await page.selectOption('select#service', { label: 'Outpatient Department' });
+  });
+
+  await test.step('And I make appointment as “Scheduled”', async () => {
+    await page.getByLabel('Select an appointment type').selectOption('Scheduled');
+  });
+
+  await test.step('And I set date for tomorrow', async () => {
+    const tomorrow = new Date();
+    tomorrow.setDate(tomorrow.getDate() + 1);
+    await page.fill('input[placeholder="dd/mm/yyyy"]', tomorrow.toLocaleDateString('en-GB'));
+  });
+
+  await test.step('And I set the “Duration” to 60', async () => {
+    await page.getByLabel('Duration (minutes)').fill('60');
+  });
+
+  await test.step('And I add a note', async () => {
+    await page.getByPlaceholder(/Write any additional points here/i).fill('Testing Appointments notes');
+  });
+
+  await test.step('And I click Save button', async () => {
+    await page.getByRole('button', { name: /save and close/i }).click();
+  });
+
+  await test.step('Then I should see a success message', async () => {
+    await expect(page.getByText(/Appointment scheduled/i)).toBeVisible();
+  });
+
+  await test.step('When I click the options kebab menu in the appointment', async () => {
+    await page.getByRole('button', { name: 'Options' }).click();
+  });
+
+  await test.step('And I choose the "Edit" option from the popup menu', async () => {
+    await page.getByRole('menuitem', { name: 'Edit' }).click();
+  });
+
+  await test.step('When I change to “Inpatient ward” location', async () => {
+    await page.selectOption('select#service', { label: 'General Medicine service' });
+  });
+
+  await test.step('And I change to “General Medicine” Service', async () => {
+    await page.getByLabel('Select a service').selectOption('General Medicine service');
+  });
+
+  await test.step('And I change the date to Today', async () => {
+    const today = new Date();
+    today.setDate(today.getDate());
+    await page.fill('input[placeholder="dd/mm/yyyy"]', today.toLocaleDateString('en-GB'));
+  });
+
+  await test.step('And I set the “Duration” of the appointment”', async () => {
+    await page.getByLabel('Duration (minutes)').fill('80');
+  });
+
+  await test.step('And I change the note', async () => {
+    await page.getByPlaceholder(/Write any additional points here/i).fill('Editing Appointmentments notes');
+  });
+
+  await test.step('And I click Save button', async () => {
+    await page.getByRole('button', { name: /save and close/i }).click();
+  });
+
+  await test.step('Then I should see a success message', async () => {
+    await expect(page.getByText(/Appointment edited/i)).toBeVisible();
+  });
+
+  await test.step('When I click the options kebab menu in the appointment', async () => {
+    await page.getByRole('button', { name: 'Options' }).click();
+  });
```

Comment:
Replace the above step with the steps below.
```js
  await test.step('When I click the options kebab menu in the appointment', async () => {
    await page.getByRole('tab', { name: 'Today' }).click();
    await page.getByRole('button', { name: 'Options' }).click();
  });
```
of-course you will need to refactor the each of the actions to be a separate step definition....

& fix the typo in the step title.

Reply:
Typo issue? 
Here?
`When I click the options kebab menu in the appointment`

Reply:
Simply use ...
```suggestion
  When I click the options menu in the appointments table
```

Reply:
BTW @Vijaykv5  what is the meaning of the word `kebab`?

Reply:
It's just an [Icon](https://www.computerhope.com/jargon/k/kebab-menu.htm) name in UI terms

Reply:
Thanks @kdaud I think that fixes the issue

Reply:
Are the changes passing locally on the local instance?

Reply:
Yes it does!